.github/workflows/ci.yml

```diff
@@ -58,13 +58,127 @@ jobs:
           secret_vault_roleId: ${{ secrets.VAULT_ROLE_ID }}
           secret_vault_secretId: ${{ secrets.VAULT_SECRET_ID }}
 
+  unit-tests:
+    if: needs.detect-changes.outputs.actionlint == 'true'
```

Comment:
Reusing the `needs.detect-changes.outputs.actionlint` change detection condition is most likely not what you want, it only triggers when GHA files change. Need to introduce a new condition in `.github/actions/paths-filter`.

For which file changes does this job need to be executed?

What I used in the POC: https://github.com/camunda/camunda/pull/17826/files

Reply:
I can help with the implementation of the filter if you tell me the requirements.

Reply:
TBH everytime when code changes 🤔

Reply:
> TBH everytime when code changes 🤔

Yea fair point, but still then docs != code so we can avoid running this part of the CI for docs-only changes e.g.. Identifying which folders/code exactly are relevant to the job is something we should do.

Reply:
> Whcih is why I think the would be helpful https://maven.apache.org/extensions/maven-build-cache-extension/

Makes sense to use this inside the job I think - but I have no experience yet with it so starting with path-based filtering is my first approach to filter on a higher-level.